src/connection.h

```diff
@@ -142,22 +120,20 @@ class Connection : public std::enable_shared_from_this<Connection>
 
 		std::recursive_mutex m_connectionLock;
 
+		std::queue<OutputMessage_ptr> messageQueue;
```

Comment:
It does in the cases where it doesn't have to grow the internal vectors, so it's amortized constant time insertions, compared to constant time with list. If you prefer a deque, you may switch to that, it has the clear() method as well.